sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzePartitionCommand.scala

```diff
@@ -0,0 +1,149 @@
+      val tableId = table.identifier
+      throw new AnalysisException(s"Partition specification for table '${tableId.table}' " +
+        s"in database '${tableId.database}' refers to unknown partition column(s): " +
+        invalidColumnNames.mkString(","))
+    }
+
+    val filteredSpec = partitionSpec.filter(_._2.isDefined)
+    if (filteredSpec.isEmpty) {
+      None
+    } else {
+      Some(filteredSpec.mapValues(_.get))
+    }
+  }
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val sessionState = sparkSession.sessionState
+    val db = tableIdent.database.getOrElse(sessionState.catalog.getCurrentDatabase)
+    val tableIdentWithDB = TableIdentifier(tableIdent.table, Some(db))
+    val tableMeta = sessionState.catalog.getTableMetadata(tableIdentWithDB)
+    if (tableMeta.tableType == CatalogTableType.VIEW) {
+      throw new AnalysisException("ANALYZE TABLE is not supported on views.")
+    }
+
+    val partitionValueSpec = validatePartitionSpec(tableMeta)
+
+    val partitions = sessionState.catalog.listPartitions(tableMeta.identifier, partitionValueSpec)
+
+    if (partitions.isEmpty) {
+      if (partitionValueSpec.isDefined) {
+        throw new NoSuchPartitionException(db, tableIdent.table, partitionValueSpec.get)
+      } else {
+        // the user requested to analyze all partitions for a table which has no partitions
+        // return normally, since there is nothing to do
+        return Seq.empty[Row]
+      }
+    }
+
+    // Compute statistics for individual partitions
+    val rowCounts: Map[TablePartitionSpec, BigInt] =
+      if (noscan) {
+        Map.empty
+      } else {
+        calculateRowCountsPerPartition(sparkSession, tableMeta, partitionValueSpec)
+      }
+
+    // Update the metastore if newly computed statistics are different from those
+    // recorded in the metastore.
+    val partitionStats = partitions.map { p =>
+      val newTotalSize = CommandUtils.calculateLocationSize(sessionState,
+        tableMeta.identifier, p.storage.locationUri)
+      val newRowCount = rowCounts.get(p.spec)
+      val newStats = CommandUtils.compareAndGetNewStats(tableMeta.stats, newTotalSize, newRowCount)
+      (p, newStats)
+    }
+
+    val newPartitions = partitionStats.filter(_._2.isDefined).map { case (p, newStats) =>
+      p.copy(stats = newStats)
+    }.toList
+
+    if (newPartitions.nonEmpty) {
+      sessionState.catalog.alterPartitions(tableMeta.identifier, newPartitions)
+    }
+
+    Seq.empty[Row]
+  }
+
+  private def calculateRowCountsPerPartition(
+      sparkSession: SparkSession,
+      tableMeta: CatalogTable,
+      partitionValueSpec: Option[TablePartitionSpec]): Map[TablePartitionSpec, BigInt] = {
+    val filter = if (partitionValueSpec.isDefined) {
+      val filters = partitionValueSpec.get.map {
+        case (columnName, value) => EqualTo(UnresolvedAttribute(columnName), Literal(value))
+      }
+      Some(filters.reduce(And))
+    } else {
+      None
```

Comment:
@gatorsmile , I don't understand this comment. Do you want me to change None to Literal.TrueLiteral? This doesn't compile and doesn't make sense to me as Literal.TrueLiteral just represents 'true' which is not a valid filter. Could you explain a little bit?